### PR TITLE
#164798018 User can get another users profile

### PR DESCRIPTION
**Pull request by sojida**
#### Description
This PR is a get request route to get a users profile

#### Type of change
- [x] New feature(create route to get user profile)

#### How Has This Been Tested?
- [x] Unit testing

#### Checklist:
- [x] Write test cases
- [x] Create routes to get user profile

#### PT-ID
[#164798018](https://www.pivotaltracker.com/story/show/164798018)

#### Screenshot
Screenshot 2019-04-02 at 9 31 23 AM

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
+import '@babel/polyfill';
```
Comment (rotimi-babalola) [inline server/app.js]: TODO: Get back to you about fixing this

File: server/app.js [hunk 2]
```diff
@@ -12,6 +14,8 @@ app.get('/', (req, res) => {
   res.send('Welcone to Authors Heaven');
 });
```
Comment (rotimi-babalola) [inline server/app.js:14]: Nitpick: Typo should be `Welcome`

File: server/routes/index.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import express from 'express';
+import articleRoute from './profile.routes';
```
Comment (rotimi-babalola) [inline server/routes/index.js]: This should be `profileRoutes`

File: server/middlewares/profile.middleware.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+   * @returns {object} error for user not found
+   * @returns {function} next function
+   */
+  async checkIfUserExist(req, res, next) {
+    if (!validations.verifyUUID(req.params.id)) {
+      return res.status(400).json({
+        error: 'id not valid',
+      });
+    }
+
```
Comment (rotimi-babalola) [inline server/middlewares/profile.middleware.js]: This should be in the profile controller. You don't need this `checkIfUserExist` middleware

File: server/helpers/profiler.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+ * @returns {object} Necessary user profile details
+ */
+const getProfile = userData => {
+  delete userData.dataValues.password;
+  delete userData.dataValues.is_activated;
+
+  return userData;
+};
+
+export default getProfile;
```
Comment (rotimi-babalola) [inline server/helpers/profiler.js]: You don't need this helper function to delete these values. Use the options I showed you on the call

**Review by rotimi-babalola — CHANGES_REQUESTED** [review]

File: server/models/rating.js [hunk 6]
```diff
@@ -11,7 +11,7 @@ module.exports = (sequelize, DataTypes) => {
       primaryKey: true,
     },
     rating_value: {
-      type: DataTypes.integer,
+      type: DataTypes.INTEGER,
```
Comment (rotimi-babalola) [inline server/models/rating.js:14]: I noticed you sneaked this in 👀

**Review by rotimi-babalola — COMMENTED** [review]
I'm just curious - are we currently checking for tokens in the request to get a user's profile by id? I can't see anything to suggest that from what I see here. @sojida

File: server/models/rating.js (hunk 6 above)
Comment (sojida) [reply server/models/rating.js:14]: When I pushed I noticed this typo was causing an error in travis

Comment (rotimi-babalola) [reply server/models/rating.js:14]: No worries 🙂

**Comment by sojida** [comment]
> I'm just curious - are we currently checking for tokens in the request to get a user's profile by id? I can't see anything to suggest that from what I see here. @sojida

@rotimi-babalola I am not checking for tokens yet because my teammate has not implemented that yet. When they do we would update the route with that function

**Comment by rotimi-babalola** [comment]
> > I'm just curious - are we currently checking for tokens in the request to get a user's profile by id? I can't see anything to suggest that from what I see here. @sojida
> 
> @rotimi-babalola we are not checking for tokens yet because my teammate has not implemented that yet. When they do we would update the route with that function

Ok. I'll merge this @sojida when that other task is done I will be expecting another PR to add ☝️ functionality

**Pull request merged**